InterviewQuestionsTests/Chapters/CustomLinkedLists/CustomLinkedList.Tests.cs

```diff
@@ -1,28 +1,40 @@
-﻿using InterviewQuestions.Chapters.CustomLinkedLists;
-
-namespace InterviewQuestionsTests.Chapters.CustomLinkedLists
-{
-    [TestClass]
-
-    public class CustomLinkedListTests
-    {
-        [TestMethod]
-        public void CustomLinkedListAddTest()
-        {
-            CustomLinkedList list = new CustomLinkedList();
-
-            list.add(10);
-            list.add(11);
-            list.add(12);
-            list.add(13);
-
-            Assert.AreEqual(list?.head?.data, 10);
-            Assert.AreEqual(list?.head?.next?.data, 11);
-            Assert.AreEqual(list?.head?.next?.next?.data, 12);
-            Assert.AreEqual(list?.head?.next?.next?.next?.data, 13);
-            Assert.AreEqual(list?.head?.next?.next?.next?.next, null);
-        }
-
-
-    }
-}
+﻿using InterviewQuestions.Chapters.CustomLinkedLists;
+
+namespace InterviewQuestionsTests.Chapters.CustomLinkedLists
+{
+    [TestClass]
+
+    public class CustomLinkedListTests
+    {
+        [TestMethod]
+        public void CustomLinkedListAddTest()
+        {
+            CustomLinkedList list = new CustomLinkedList();
+
+            list.add(10);
+            list.add(11);
+            list.add(12);
+            list.add(13);
+
+            Assert.AreEqual(list?.head?.data, 10);
+            Assert.AreEqual(list?.head?.next?.data, 11);
+            Assert.AreEqual(list?.head?.next?.next?.data, 12);
+            Assert.AreEqual(list?.head?.next?.next?.next?.data, 13);
+            Assert.AreEqual(list?.head?.next?.next?.next?.next, null);
+        }
+        [TestMethod]
+        public void CustomLinkedListReturnValueTest() 
+        {
+            CustomLinkedList list = new CustomLinkedList();
+            list.add(1);
+            list.add(2);
+            list.add(3);
+            list.add(4);
+            list.add(5);
+
+            Assert.AreEqual(list.getKhtLastElement(1), 5);
+            Assert.AreEqual(list.getKhtLastElement(3), 3);
+            Assert.AreEqual(list.getKhtLastElement(5), 1);
+        }

```

Comment:
check out of boundaries as well. method should not chrash

Reply:
i do not get it exactly what to do?

Reply:
let's say you have a linked list whose length is 5.
what happens if I call list.getKhtLastElement(7) ??
so add a new unit test that checks the out of the boundaries as I said above, and be sure that expected exceptions are thrown.

**Hint**: search how to unit test expected exceptions in c#